src/lbaf/IO/lbsMeshWriter.py

```diff
@@ -0,0 +1,308 @@
+        if streamer.Error:
+            self.__logger.error(f"Failed to instantiate a grid streamer for file {self.__rank_file_name}")
+            sys.exit(1)
+        else:
+            self.__logger.info(f"Writing ExodusII file: {self.__rank_file_name}")
+            writer = vtk.vtkExodusIIWriter()
+            writer.SetFileName(self.__rank_file_name)
+            writer.SetInputConnection(streamer.Algorithm.GetOutputPort())
+            writer.WriteAllTimeStepsOn()
+            writer.Update()
+
+    def write_object_view_file(self, distributions: dict):
+        """ Map objects to grid and write ExodusII file
+        """
+
+        # Retrieve number of mesh points and bail out early if empty set
+        n_o = self.__phase.get_number_of_objects()
+        if not n_o:
+            self.__logger.error("Empty list of objects, cannot write a mesh file")
+            return
+
+        # Number of edges is fixed due to vtkExodusIIWriter limitation
+        n_e = int(n_o * (n_o - 1) / 2)
+        self.__logger.info(f"Creating object view mesh with {n_o} points, {n_e} edges, and jitter factor: {self.__object_jitter}")
+
+        # Determine available dimensions for object placement in ranks
+        rank_dims = [d for d in range(3) if self.__grid_size[d] > 1]
+
+        # Iterate over all object distributions
+        for iteration, object_mapping in enumerate(distributions["objects"]):
+            # Create point array for object times
+            t_arr = vtk.vtkDoubleArray()
+            t_arr.SetName("Time")
+            t_arr.SetNumberOfTuples(n_o)
+
+            # Iterate over ranks and objects to create mesh points
+            points = vtk.vtkPoints()
+            points.SetNumberOfPoints(n_o)
+            point_to_index = {}
+            index_to_point = []
+            point_index = 0
+            sent_volumes = []
+            for rank_id, objects in enumerate(object_mapping):
+                # Determine rank offsets
+                offsets = [
+                    self.__grid_resolution * c
+                    for c in self.global_id_to_cartesian(rank_id, self.__grid_size)]
+
+                # Iterate over objects and create point coordinates
+                n_o_rank = len(objects)
+                n_o_per_dim = math.ceil(n_o_rank ** (1. / len(rank_dims)))
+                self.__logger.debug(f"Arranging a maximum of {n_o_per_dim} objects per dimension in {rank_dims}")
+                o_resolution = self.__grid_resolution / (n_o_per_dim + 1.)
+                rank_size = [n_o_per_dim if d in rank_dims else 1 for d in range(3)]
+                centering = [0.5 * o_resolution * (n_o_per_dim - 1.)
+                             if d in rank_dims else 0.0 for d in range(3)]
+                    
+                for i, o in enumerate(objects):
+                    # Insert point using offset and rank coordinates
+                    jitter = [
+                        (random.random() - 0.5) * o_resolution * self.__object_jitter
```

Comment:
Can the `random.random()` part of the jitter be computed much earlier on and stored as a property of the object itself, independent of the rank on which the object exists?

Reply:
Done!